lib/logstash/filters/aggregate.rb

```diff
@@ -202,13 +216,20 @@ def filter(event)
 			map = aggregate_maps_element.map
 
 			# execute the code to read/update map and event
-			@codeblock.call(event, map)
+			begin
+				@codeblock.call(event, map)
+				noError = true
+			rescue => exception
+				@logger.error("Aggregate exception occurred. Error: #{exception} ; Code: #{@code} ; Map: #{map} ; EventData: #{event.instance_variable_get('@data')}")
+				event.tag("_aggregateexception")
+			end
 			
 			# delete the map if task is ended
 			@@aggregate_maps.delete(task_id) if @end_of_task
 		end
 
-		filter_matched(event)
+		# match the filter, only if no error occurred
+		filter_matched(event) if noError
```

Comment:
Check if the event has the _aggregateerror tag may be a bad idea in some case.
I mean : if the event passed several aggregate filters, and a previous aggregate filter raised an error, but not current one, then the event has the _aggregateerror tag but we still want to call filter_matched() method.

What do you think about it ?